frontend/packages/dev-console/src/components/health-checks/HealthCheckProbe.tsx

```diff
@@ -0,0 +1,108 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { Formik } from 'formik';
+import { PlusCircleIcon, MinusCircleIcon } from '@patternfly/react-icons';
+import { Button, ButtonVariant } from '@patternfly/react-core';
+import ProbeForm from './ProbeForm';
+import { HealthCheckProbeDataType } from './health-checks-types';
+import { getHealthChecksProbe } from './health-check-probe-utils';
+import { getValidationSchema } from './health-check-probe-validation-utils';
+import './HealthCheckProbe.scss';
+
+interface HealthCheckProbeProps {
+  probe: string;
+  initialValues: HealthCheckProbeDataType;
+}
+
+const HealthCheckProbe: React.FC<HealthCheckProbeProps> = ({ probe, initialValues }) => {
+  const [addProbe, setAddProbe] = React.useState(false);
+  const [selectedProbe, setSelectedProbe] = React.useState<React.ReactNode>('');
+  const [probeData, setProbeData] = React.useState<HealthCheckProbeDataType>();
+
+  const onEditProbe = () => {
+    setAddProbe(true);
+  };
+
+  const handleDeleteProbe = () => {
+    setAddProbe(false);
+    setSelectedProbe('');
+    setProbeData(initialValues);
+  };
+
+  const handleReset = (values, actions) => {
+    actions.resetForm({ values: initialValues, status: {} });
+    setAddProbe(false);
+  };
+
+  const handleSubmit = (values, actions) => {
+    actions.setSubmitting(true);
+    setAddProbe(false);
+    setProbeData(values);
+    setSelectedProbe(
+      <>
+        {`${getHealthChecksProbe(probe).formTitle} Added`}
+        <Button
+          className="pf-m-plain--align-left"
+          variant={ButtonVariant.plain}
+          onClick={handleDeleteProbe}
+        >
+          <MinusCircleIcon />
+        </Button>
+      </>,
+    );
+  };
+
+  const handleAddProbe = () => {
+    setAddProbe(true);
+  };
+
+  const renderProbe = () => {
+    if (addProbe) {
+      return (
+        <Formik
+          initialValues={!_.isEmpty(probeData) ? probeData : initialValues}
```

Comment:
So your `initialValues` changes based on the user entered values? What does that do exactly?

Reply:
I did this to preview the entered values when user clicks on `Edit Probe`

Reply:
That is the issue, you're using wrong patterns to do these things. `initialValues` of the form should not change like that.